### PR TITLE
Remove RichEmbed in favour of MessageEmbed

### PR DESCRIPTION
**Pull request by Drahcirius**
From #1529, MessageEmbeds are now identical to RichEmbeds (with a few differences).
There's no further reason in having separate MessageEmbed and RichEmbed classes, as they're similar enough that you could require MessageEmbed inside of RichEmbed and export it to get identical behaviour. 

This simplifies internal usage as well. Typechecking #1451 for RichEmbed and MessageEmbed could be simplified to just MessageEmbed. 

In this commit:
`<MessageEmbed>._apiTransform()` private method added so that internal variables can continue using camel case.
`<MessageEmbed>.color` getter returns null if colour is not set instead of throwing an error.
`<MessageEmbed>.createdAt` getter returns null if timestamp is unset/invalid.
`<RichEmbed>.addBlankField ` added to MessageEmbed.

Most properties that used to to be set to null when missing now default to undefined, to be more consistent with current RichEmbed behaviour.

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/MessageEmbed.js [hunk 1]
```diff
@@ -118,10 +114,7 @@ class MessageEmbed {
      * @property {string} name The name of this provider
      * @property {string} url URL of this provider
      */
-    this.provider = data.provider ? {
-      name: data.provider.name,
-      url: data.provider.url,
-    } : null;
+    this.provder = data.provider;
```
Comment (kyranet) [inline src/structures/MessageEmbed.js]: Nice typo?

Comment (GloriousYellow) [inline src/structures/MessageEmbed.js]: There are unit tests, right? Won't they catch it?

**Comment by devsnek** [comment]
can we just get rid of addBlankField

File: src/structures/MessageEmbed.js [hunk 2]
```diff
@@ -177,6 +168,15 @@ class MessageEmbed {
   }
 
   /**
+   * Convenience function for `<MessageEmbed>.addField('\u200B', '\u200B', inline)`.
+   * @param {boolean} [inline=false] Set the field to display inline
+   * @returns {MessageEmbed} This embed
+   */
+  addBlankField(inline = false) {
```
Comment (devsnek) [inline src/structures/MessageEmbed.js]: delet

Comment (iCrawl) [inline src/structures/MessageEmbed.js]: No.

Comment (KingDGrizzle) [inline src/structures/MessageEmbed.js]: ~Why? Its such a useful small thing! Ain't nobody got time to get the unicode code for the 0-width space~

**Pull request merged**